code/entities/weapons/fnb/GooglyEye.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Sandbox;
+
+namespace Cinema;
+
+public partial class GooglyEye : WeaponBase
+{
+    static private float MaxPlacementDistance = 96.0f;
+    PreviewEntity PreviewModel { get; set; }
+
+    protected virtual bool IsPreviewTraceValid(TraceResult tr)
+    {
+        if (!tr.Hit || !tr.Entity.IsValid() || tr.Entity is GooglyEyeEntity) return false;
+
+        return true;
+    }
+    public override void Simulate(IClient cl)
+    {
+        base.Simulate(cl);
+
+        if (Game.IsClient) return;
+        
+        using (Prediction.Off())
+        {
+            if (!Input.Pressed("attack1")) return;
+               
+            var ray = Owner.AimRay;
+            var distance = MaxPlacementDistance;
+            var tr = Trace.Ray(ray, distance)
```

Comment:
This isnt bad, however without having a look at how the distance works internally i can recommend using the startpos of the aimray and then using AimRay.Forward * Distance to only do the trace for the distance it can be.

If you know that the way it is now is as optimized then just keep it :)